cdmtaskservice/app_state.py

```diff
@@ -86,3 +96,22 @@ def _get_app_state_from_app(app: FastAPI) -> AppState:
     if not app.state._cdmstate:
         raise ValueError("App state has not been initialized")
     return app.state._cdmstate
+
+
+async def get_mongo_client(cfg: CDMTaskServiceConfig) -> AsyncIOMotorClient:
+    client = AsyncIOMotorClient(
+        cfg.mongo_host,
+        # Note auth is only currently tested manually
+        authSource=cfg.mongo_db,
+        username=cfg.mongo_user,
+        password=cfg.mongo_pwd,
+        retryWrites=cfg.mongo_retrywrites,
+    )
+    # Test connnection cheaply, doesn't need auth.
+    # Just throw the exception as is
+    try:
+        await client.admin.command("ismaster")
+        return client
+    except:
+        client.close()
+        raise
```

Comment:
Do you think we should explicitly close it in a `finally` block, or is it fine to leave it open in this case?

```suggestion
    try:
        yield client
    finally:
        await client.close()
```

Reply:
I don't think you can put a yield in a try finally block: https://peps.python.org/pep-0255/#specification-yield

> Restriction: A yield statement is not allowed in the try clause of a try/finally construct. The difficulty is that there’s no guarantee the generator will ever be resumed, hence no guarantee that the finally block will ever get executed; that’s too much a violation of finally’s purpose to bear.

Reply:
I'm going to go ahead and merge but we can keep discussing I'll implement whatever we agree on in a future PR